conda/core/prefix_data.py

```diff
@@ -84,6 +100,35 @@ def __init__(
             else context.pip_interop_enabled
         )
 
+    @classmethod
+    def from_name(cls, name: str, **kwargs):
+        if "/" in name or "\\" in name:
+            raise CondaValueError("Environment names cannot contain path separators")
+        try:
+            return cls(locate_prefix_by_name(name))
+        except EnvironmentNameNotFound:
+            cls(name).validate_name()
+            return cls(Path(_first_writable_envs_dir(), name), **kwargs)
+
+    @classmethod
+    def from_context(cls, validate: bool = False):
+        inst = cls(context.target_prefix)
+        if validate:
+            inst.validate_path()
+            inst.validate_name()
+        return inst
+
+    @property
+    def name(self) -> str:
```

Comment:
It would be good to document this too if we expect others to be using it.

Another thing that would be nice would be explaining what an empty string means. I'm pretty sure it means it's not a named environment, but totally not clear by reading the code itself.